### ISSUE TITLE
Do not call sys.exc_clear() on Python 3 in twisted/scripts/trial.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8468]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8468|
|Type|enhancement|
|Created|2016-06-13 22:00:02Z|
|Branch|https://github.com/twisted/twisted/tree/8468-rodrigc-exc_clear|

sys.exc_clear() is gone on Python 3: https://docs.python.org/3/whatsnew/3.0.html#index-22

We should not try to call it on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8468 8468
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8468_rodrigc_exc_clear 8468-rodrigc-exc_clear
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465855202122172 1465855202122172
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/187

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Here's some review that applies not only to this but to [#8465](https://github.com/twisted/twisted/issues/8465), [#8467](https://github.com/twisted/twisted/issues/8467), [#8468](https://github.com/twisted/twisted/issues/8468), and [#8469](https://github.com/twisted/twisted/issues/8469): what the hell is `exc_clear()` even doing there? It seems to serve no purpose whatsoever. Glyph has suggested it has something to do with a Python 2.2 implementation detail, so I'm inclined to say this is crazy-talk.

I'm going to let this build run, but I'd like to see one of two things:

- Either more aggressive patch that simply changes the except block to `pass`.
- Or the most aggressive patch, which removes all of this nonsense and makes these all `console_scripts`, such that `pip install -e '.[dev]'` (the command we tell people to run for development anyway) makes those scripts available directly.

I'll happily accept either kind of patch, but we should definitely do one of them.

I'll apply this feedback to the other tickets as well. Feel free to ping me directly when you're ready for more review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc set owner to @Lukasa|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

I updated the branch: https://github.com/twisted/twisted/compare/trunk...8468-rodrigc-exc_clear

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set owner to @rodrigc|
|-|-|

Going in a different direction on this after talking to Lukasa

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc commented|
|-|-|

Putting back into review pile : https://github.com/twisted/twisted/pull/187

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 6e0937997ced5a2a7860afd74fb38e6ca2085205
```
#!CommitTicketReference repository="" revision="6e0937997ced5a2a7860afd74fb38e6ca2085205"
Merge pull request #344 from twisted/8468-rodrigc-exc_clear

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8468

Do not call sys.exc_clear() on Python 3 in twisted/scripts/trial.py
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
